src/qililab/platform/platform.py

```diff
@@ -640,14 +641,16 @@ def execute_anneal_program(
             averages (int, optional): Amount of times to run and average the program over. Defaults to 1.
         """
         if calibration.has_waveform(bus=readout_bus, name=measurement_name):
-            annealing_program = AnnealingProgram(self, annealing_program_dict)
+            annealing_program = AnnealingProgram(
```

Comment:
here at this point you already have the calibration class created to use as a parameter to the class AnnealingProgram.

Reply:
It is true that we require a `Calibration` for `execute_anneal_program` but so far this was not a requirement for `AnnealingProgram` itself (just considering modularity). I think it'd be better to require a `CrosstalkMatrix` from `get_waveforms`  and for `execute_anneal_program`  to load that xtalk matrix from the calibration file (just like we do for calibrated measurements)

Reply:
That way if someone wants to run methods from `AnnealProgram` as standalone they can just create their own xtalk matrix without needing to load a calibration file

Reply:
Whatever you think is best, but in the execute_anneal you have the parameter already if you need to use it, how you propagate it until the point of usage from there is to you to propose. But as it is now, you can already use a calibration object at that point.

Reply:
What I want to say is, do not leave a todo hanging in there if you already have a way to solve it. This should be a complete PR as much as possible. If in the future we re-visit decisions is ok, but do not leave a PR uncompleted to have yet another one to finish it.

Reply:
It was left as it is because at the time we did not have a method to load the calibration matrix from a calibration. Since now we have it at https://github.com/qilimanjaro-tech/qililab/pull/784 i will merge those files from there

Reply:
Sounds good.

Reply:
resolving this now.